src/System.Net.Http/tests/FunctionalTests/DiagnosticsTests.cs

```diff
@@ -503,21 +510,22 @@ public void SendAsync_ExpectedDiagnosticSynchronousExceptionActivityLogging()
                     using (HttpClientHandler handler = CreateHttpClientHandler(useSocketsHttpHandlerString))
                     using (HttpClient client = new HttpClient(handler))
                     {
+                        // Set a https proxy.
+                        handler.Proxy = new WebProxy($"https://{Guid.NewGuid()}.com", false);
+                        HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, $"http://{Guid.NewGuid()}.com");
+
                         if (bool.Parse(useSocketsHttpHandlerString))
                         {
-                            // Forces a synchronous exception for SocketsHttpHandler
-                            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, $"http://{Guid.NewGuid()}.com");
-                            request.Version = new Version(0, 0);
-
+                            // Forces a synchronous exception for SocketsHttpHandler.
+                            // SocketsHttpHandler only allow http scheme for proxies.
                             Assert.ThrowsAsync<NotSupportedException>(() => client.SendAsync(request)).Wait();
```

Comment:
I don't understand.  The comment says "force a synchronous exception", but if it actually propagates out of the call to SendAsync, then we should be using ```Assert.Throws<NotSupportedException>(() => { client.SendAsync(request); });```, so... is the comment wrong or is the assert wrong?